src/wallet/rpcwallet.cpp

```diff
@@ -2141,6 +2141,9 @@ static UniValue encryptwallet(const JSONRPCRequest& request)
     if (!pwallet->EncryptWallet(strWalletPass)) {
         throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: Failed to encrypt the wallet.");
     }
+    if (!pwallet->Unlock(strWalletPass)) {
```

Comment:
I don't understand why this change (unlocking the wallet after it's encrypted in RPC) is necessary or desirable. If you want to keep it, can you add some justification (perhaps to the commit log)?

Reply:
Dropped that commit.